lib/model/binding.dart

```diff
@@ -66,6 +68,34 @@ abstract class ZulipBinding {
     _instance = this;
   }
 
+  /// Provides device and operating system information,
+  /// via package:device_info_plus.
+  ///
+  /// The returned Future resolves to null if an error is
+  /// encountered while fetching the data.
+  Future<BaseDeviceInfo?> get deviceInfo;
+
+  /// Provides device and operating system information,
+  /// via package:device_info_plus.
+  ///
+  /// May return null if prefetching hasn't completed yet,
+  /// or an error occured while fetching the data.
+  BaseDeviceInfo? get maybeDeviceInfo;
```

Comment:
I feel like "maybe" doesn't quite capture the relationship between this and `deviceInfo`. We often say "foo" and "maybeFoo" when the former throws in the absence of a result and the latter returns null, but this is something else.

How about:
```suggestion
  BaseDeviceInfo? get syncDeviceInfo;
```

I think that fairly well captures why you might want this one. And then it's natural that the trade-off is that the sync version might not have data yet.